src/bitcoind.ts

```diff
@@ -1,5 +1,146 @@
+import Client from "bitcoin-core"
 import _ from "lodash"
-import { BitcoindClient, bitcoindDefaultClient, btc2sat } from "./utils"
+import { btc2sat } from "./utils"
+
+const connection_obj = {
+  network: process.env.NETWORK,
+  username: "rpcuser",
+  password: process.env.BITCOINDRPCPASS,
+  host: process.env.BITCOINDADDR,
+  port: process.env.BITCOINDPORT,
+  version: "0.21.0",
+}
+
+type ScriptPubKey = {
+  asm: string
+  hex: string
+  reqSigs: number
+  type: string
+  addresses: [string]
+}
+
+type VOut = {
+  value: number
+  n: number
+  scriptPubKey: ScriptPubKey
+}
+
+type GetAddressInfoResult = {
+  address: string
+  // TODO
+}
+
+type InWalletTransaction = {
+  amount: number
+  fee: number
+  // TODO
+  txid: string
+}
+
+// TODO consistency in params
+
+export class BitcoindClientNoWallet {
+  client
+
+  constructor() {
+    this.client = new Client({ ...connection_obj })
+  }
+
+  async getBlockCount(): Promise<number> {
+    return await this.client.getBlockCount()
+  }
+
+  async getBlockchainInfo(): Promise<{ chain: string }> {
+    return await this.client.getBlockchainInfo()
+  }
+
+  async createWallet({
+    wallet_name,
+  }: {
+    wallet_name: string
+  }): Promise<{ name: string; warning: string }> {
+    return await this.client.createWallet({ wallet_name })
+  }
+
+  async listWallets(): Promise<[string]> {
+    return await this.client.listWallets()
+  }
+
+  async loadWallet(filename: string): Promise<{ name: string; warning: string }> {
```

Comment:
Shouldn't the `xxxWallet` related functions be part of the `BitcoindWalletClient`?